ecommerce/products/templates/product_list.html

```diff
@@ -0,0 +1,16 @@
+<ul>
+    <li>
+        <div>
+            <a href="{% url 'detail' product.id %}">
+                <div><strong>Name: </strong>{{ product.name }}</div>
+            </a>
+            <a href="{% url 'product-by-type' product.product_type %}">
+                <div><strong>Type: </strong>{{ product.product_type }}</div>
+            </a>
+            <div><strong>Language: </strong>{{ product.language }}</div>
+            {% if product.rarity %}
+            <div><strong>Rarity: </strong>{{ product.rarity }}</div>
+            {% endif %}
+        </div>
+    </li>
+</ul>
```

Comment:
I think you should make a sub directory inside `/templates` called `/components` where you keep this type of template. Should help keep things clearer and more organised.